PowerEditor/src/NppIO.cpp

```diff
@@ -2140,7 +2140,17 @@ bool Notepad_plus::loadSession(Session & session, bool isSnapshotMode, bool shou
 		}
 		else
 		{
-			lastOpened = BUFFER_INVALID;
+			// file doesn't exist and has no backup; probably it's a file that was deleted while NPP was closed
+			// as a hack to ensure that the file can be restored without issues, we will create an empty file
+			// with that name, open it normally, and then delete it. This way the user gets a missing file prompt for the file.
+			if (!PathFileExists(pFn) && MainFileManager.createEmptyFile(pFn))

```

Comment:
If we arrive here, PathFileExists(pFn) is surely FALSE, `!PathFileExists(pFn)` is not needed here and it shouldn't be used again.

Reply:
But `we will create an empty file with that name, open it normally, and then delete it. This way the user gets a missing file prompt for the file.` is a very bad workaround which shouldn't even be considered.